cves/CVE-2016-4461.yml

```diff
@@ -61,31 +72,35 @@ fixes_vcc_instructions: |
   Please put the SVN commit number in "commit" below, and any notes about how this
   was discovered in the "note" field.
 fixes:
-  - commit: 46517afb1402fbf3fd84089920b43cc07e948a92
-    note: Added missing blocked classes. Struts 2.3.29
-  - commit: 5190b53673a710ead31bbb5f82cf4ca171994629
-    note: Blocks chained expressions
-  - commit: d36f31b3e1a65c619b1cbaf743baa3efc66168be
-    note: Upgrades OGNL
+  - commit: 9ac863b339a3513dabd417f4be8a802418a997ba
+    note: Introduces more restrictive SMI (Strict Method Invocation) - ActionConfig.java
+  - commit: 016b8eedd5ca47fce95e2a16069e640e3842906d
+    note: Adds additional blocked classes - struts-default.xml
+  - commit: f238cf4f1091be19fbcfd086b042c86a1bcaa7fc
+    note: Uses isSequence flag to block chained expressions - OgnlUtil.java
+  - commit: 27ca165ddbf81c84bafbd083b99a18d89cc49ca7
+    note: Throws away methods that don't match pattern for allowed actions - DefaultActionMapper.java
+  - commit: 9b34a28190b71f73c26564ce06139aa4e4861e18
+    note: Disallow dot (.) in action name - DefaultActionMapper.java
+  - commit: 7da4ef39023bb5d86509d65565a04b533e8b4c55
+    note: Introduces new callMethod() function to execute actions (cherry pick commit b28b78) - DefaultActionInvocation.java, OgnlUtil.java
 vccs:
-  - commit: 51b276c2d80c4008b86f47a0232c32e8385c0096
-    note: Implemented the duplicate access. Apache 2.3.20.2
-  - commit: d92654daf3dfe36082995b713484de92041afaea
-    note: Implemented the duplicate access. Apache 2.3.24.2
-  - commit: fbb91d1bb9b7c8033d8de02f19f75daae192ce2f
-    note: Reverted Excluded classes. Apache 2.3.28
-  - commit:
-  - note:
+  - commit: f54afdc3f441dc48d012aebd3cbbdcd542811cee
+    note: Initial implementaiton for allowed methods when executing OGNL actions. Probable root cause component of predecessor vulnerability CVE-2016-0785 due to lackluster input sanitization.
+  - commit: 61a7ee296161bbfa61e90871649598a2e4a680a2 
+    note: Sign of early efforts to mitigate double evaluation by disabling the evaluation expression altogether. This underscores issues with OGNL's evaluation expression during S2-029. 
+  - commit: 15857a69e7baf3675804495a5954cd0756ac8364
+    note: Enables expression evaluation and storage without input validation 
 
```

Comment:
Just asking for context, were all the fixes/vccs simply wrong?

Reply:
Within the original document? Yeah, pretty much.

Reply:
Yup